noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/private_kernel_circuit_public_inputs_composer.nr

```diff
@@ -220,4 +224,11 @@ impl PrivateKernelCircuitPublicInputsComposer {
             self.public_inputs.public_teardown_call_request = call_request;
         }
     }
+
+    fn propagate_fee_payer(&mut self, source: DataSource) {
+        if (source.private_call_public_inputs.is_fee_payer) {
+            assert(self.public_inputs.fee_payer.is_zero(), "Cannot overwrite non-empty fee_payer");
+            self.public_inputs.fee_payer = source.private_call_public_inputs.call_context.storage_contract_address;
```

Comment:
This should be `contract_address`?

Reply:
Exactly, caught it as well while rebasing.